Plot optimization

gdiazc opened the pull request:
The Elasticsearch instance now stores pre-computed "mean hourly sentiment ticks" in a separate index. This pull requests migrates the dashboard over to this pre-computed data so it doesn't have to be calculated on the fly.

gdiazc committed:
Remove .bs_timeline_empty class as soon as request was dispatched to
server, to avoid further requests.

gdiazc committed:
Merge branch 'development'

gdiazc committed:
Merge branch 'inhouse-sentiments'

gdiazc committed:
Retrieve mean sentiment from cached version.

bitcoin_sentiments.js:
- Add service to retrieve mean_sentiment_ticks
- Load quick plot

index.html:
- Add quick plot div

app.py:
- Add new endpoint for quick plot

es_client.py:
- Add method for retrieving ticks

gdiazc committed:
Change initial view window.

gdiazc committed:
Remove slow plot.

gdiazc committed:
Merge branch 'development'

gdiazc committed:
Merge branch 'master' of https://git.heroku.com/bitcoin-sentiment-dashboard

dev1-fund3 commented on app.py:
```diff
@@ -98,5 +98,28 @@ def get_tweets():
 
 
+@app.route('/get_mean_sentiments.json', methods=['GET'])
+def get_mean_sentiments():
```
add docstring comments for what this method does

gdiazc replied:
Fixed in [3fd135b](https://github.com/fund3/sentiment-dashboard/pull/5/commits/3fd135bb0e6622d2d74fb1c882b41c9378f306c3)

dev1-fund3 commented on es_client.py:
```diff
@@ -70,3 +69,32 @@ def get_tweets(client, max_tweets=500):
+    """
+    Get mean hourly sentiment ticks from the Elasticsearch endpoint specified by client.
+
+    :param client: Elasticsearch client object.
```
add docstring for max_ticks

gdiazc replied:
Fixed in [3fd135b](https://github.com/fund3/sentiment-dashboard/pull/5/commits/3fd135bb0e6622d2d74fb1c882b41c9378f306c3)

dev1-fund3 commented on es_client.py:
```diff
@@ -70,3 +69,32 @@ def get_tweets(client, max_tweets=500):
+        .sort({'timestamp': {'order': 'desc'}})[:max_ticks]
+    search.execute()
+
+    ticks = []
```
```suggestion
    return [
        {
            'timestamp': hit.to_dict().get('timestamp', None),
            'value': hit.to_dict().get('value', None)
        } for hit in search
    ]
```

dev1-fund3 replied:
Use list compression and fewer local vars for speed up.

gdiazc replied:
Fixed in [3fd135b](https://github.com/fund3/sentiment-dashboard/pull/5/commits/3fd135bb0e6622d2d74fb1c882b41c9378f306c3)

dev1-fund3 commented on plotting.py:
```diff
@@ -33,6 +35,8 @@ def plot_polarity_vs_time(df, polarity_column='polarity'):
 
     # Format view range:
+    now = datetime.datetime.now()
+    p.x_range = bokeh.models.Range1d(now - datetime.timedelta(days=3), now + datetime.timedelta(hours=6))
```
add a low priority TODO to make the range able to be input by user via dropdown menu.

gdiazc replied:
Fixed in [3fd135b](https://github.com/fund3/sentiment-dashboard/pull/5/commits/3fd135bb0e6622d2d74fb1c882b41c9378f306c3)

dev1-fund3 commented on static/bitcoin_sentiments.js:
```diff
@@ -26,13 +32,22 @@ angular.module('BitcoinSentimentsApp', [])
             ]
 
-            BitcoinSentimentsService.getTweetsFromServer()
+            // Get sentiment ticks quicker:
```
```suggestion
            // Get precomputed sentiment ticks:
```

gdiazc replied:
Fixed in [3fd135b](https://github.com/fund3/sentiment-dashboard/pull/5/commits/3fd135bb0e6622d2d74fb1c882b41c9378f306c3)

gdiazc committed:
update .gitignore

gdiazc committed:
Show in-house sentiments for individual tweets.

index.html:
- Show ih_sentiment property for individual tweets

app.py:
- Refactor get_tweets (now only returns list of tweets; no plot)

gdiazc committed:
Add readme.

gdiazc committed:
Removed unused imports.

gdiazc committed:
Change UI label.

gdiazc committed:
Add author_followers and retweet_count info.

gdiazc committed:
Code review modifications.
